src/permission/PermissionAttachment.php

```diff
@@ -81,29 +80,8 @@ public function clearPermissions() : void{
 		$this->recalculatePermissibles();
 	}
 
-	/**
-	 * @param bool[] $permissions
-	 * @phpstan-param array<string, bool> $permissions
-	 */
-	public function setPermissions(array $permissions) : void{
-		foreach(Utils::stringifyKeys($permissions) as $key => $value){
-			$this->permissions[$key] = $value;
-		}
-		$this->recalculatePermissibles();
-	}
-
-	/**
-	 * @param string[] $permissions
-	 */
-	public function unsetPermissions(array $permissions) : void{
-		foreach($permissions as $node){
-			unset($this->permissions[$node]);
-		}
-		$this->recalculatePermissibles();
-	}
```

Comment:
Not sure why this is removed

Reply:
These are the same reasons as above. We remove string usage for Permission so we can't set/remove multiple Permissions just we they name, we need the Permission object. It's can be replace by for-each loop by call `setPermission`/`unsetPermission`

Reply:
`unsetPermissions()` isn't affected by this.

Reply:
why ? I think it's illogic to remove usage of string but not here

Reply:
There's no need to remove it since there's no technical reason to.